geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloClientConfig.scala

```diff
@@ -0,0 +1,130 @@
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.accumulo.data
+
+import java.io.{File, FileInputStream}
+import java.util.Properties
+
+import com.typesafe.scalalogging.Logger
+import org.locationtech.geomesa.utils.io.WithClose
+import org.slf4j.LoggerFactory
+
+import scala.util.Try
+
+/**
+ * Client configuration options, loaded from a config file
+ *
+ * @param instance instance name
+ * @param zookeepers zookeeper connect string
+ * @param zkTimeout zookeeper timeout
+ * @param authType  authentication type ("password", "kerberos" or authentication token class)
+ * @param principal authentication principal (username, keytab file, etc)
+ * @param token authentication token (password, etc)
+ */
+case class AccumuloClientConfig(
+    instance: Option[String],
+    zookeepers: Option[String],
+    zkTimeout: Option[String],
+    authType: Option[String],
+    principal: Option[String],
+    token: Option[String]
+  ) {
+  override def toString: String = {
+    val values =
+      instance.map(s => s"instance=$s").toSeq ++ zookeepers.map(s => s"zookeepers=$s") ++
+        zkTimeout.map(s => s"zookeepers.timeout=$s") ++ authType.map(s => s"auth.type=$s") ++
+        principal.map(s => s"principal=$s") ++ token.map(_ => "token=***")
+    values.mkString("{", ",", "}")
+  }
+}
+
+object AccumuloClientConfig {
+
+  val PasswordAuthType = "password"
+  val KerberosAuthType = "kerberos"
+
+  private val Empty = AccumuloClientConfig(None, None, None, None, None, None)
+
+  // get the logger directly so that the logger name (and config) doesn't have a $ on the end
+  private lazy val logger: Logger = Logger(LoggerFactory.getLogger(classOf[AccumuloClientConfig]))
+
+  /**
+   * Search the classpath for Accumulo configuration files
+   *
+   * @return
+   */
+  def load(): AccumuloClientConfig = {
+    val loader = Option(Thread.currentThread().getContextClassLoader).getOrElse(getClass.getClassLoader)
+    AccumuloClientProperties(loader).orElse(ClientConf(loader)).getOrElse(Empty)
+  }
+
+  /**
+   * This class mimics the Accumulo 2.x logic for reading 'accumulo-client.properties'
+   */
+  private object AccumuloClientProperties extends ConfigLoader {
+    override protected val fileName: String = "accumulo-client.properties"
+    override protected val instanceKey: String = "instance.name"
+    override protected val zookeeperKey: String = "instance.zookeepers"
+    override protected val zookeeperTimeoutKey: String = "instance.zookeepers.timeout"
+    override protected val authTypeKey: Option[String] = Some("auth.type")
+    override protected val principalKey: Option[String] = Some("auth.principal")
+    override protected val tokenKey: Option[String] = Some("auth.token")
+  }
+
+  /**
+   * This class mimics the Accumulo 1.x logic for reading 'client.conf'
```

Comment:
Mimics or implements?

Reply:
well there's not really an API or anything, just some conventions they follow...